business/suggestionService.js

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get, set, sortBy, filter, concat } = require('lodash')
+const EntityCoordinates = require('../lib/entityCoordinates')
+
+class SuggestionService {
+  constructor(definitionService, definitionStore) {
+    this.definitionService = definitionService
+    this.definitionStore = definitionStore
+  }
+
+  /**
+   * Get suggestions for the given coordinates.
+   *
+   * @param {EntityCoordinates} coordinates - The entity for which we are looking for suggestions
+   * @returns {Suggestion} A set of suggested fixes for the definition at the given coordinates.
+   * `null` is returned if no such coordinates are found.
+   */
+  async get(coordinates) {
+    const related = await this._getRelatedDefinitions(coordinates)
+    if (!related) return null
+    const baseSuggestion = this._createBaseSuggestion(coordinates)
+    // create a struct with the strings in it and iterate instead
+    const promises = [
+      await this._collectSuggestionsForField(related, baseSuggestion, 'licensed.declared'),
+      await this._collectSuggestionsForField(related, baseSuggestion, 'described.sourceLocation'),
+      await this._collectSuggestionsForField(related, baseSuggestion, 'described.releaseDate'),
```

Comment:
releaseDate doesn't really make sense in this context. We are getting other versions of the same component, they would rarely have the same release dates

Reply:
Is the release date related to the component itself or to the version?
When implementing this i supposed that the release date would change on any new version of the same component.

Reply:
releaseDate likely does not make sense

Reply:
I was also planning to drop the file suggestions for this first iteration as we decided to leave them off of the find() results.

So that narrows this down to suggesting declared licenses and sourceLocation.